### PR TITLE
Added Bank table in the db(Included Tab)

### PR DESCRIPTION
**Pull request by eoliveros**

### CODE REVIEW

**Comment by djpnewton** [comment]
- Can we change "bank_account_number" to "account_number"
- add "bank_name"

Is "address" bank address or user address?

**Comment by eoliveros** [comment]
> * Can we change "bank_account_number" to "account_number"
> * add "bank_name"
> 
> Is "address" bank address or user address?

- changed from bank_account_number to account_number.
- address to "account_holder_address". Rename the field to something more recognisable.

File: models.py [hunk 1]
```diff
@@ -378,6 +378,38 @@ def to_json(self):
...
+
+    def generate_defaults(self):
+        self.user = current_user
+
+    @classmethod
+    def count(cls, session):
+        return session.query(cls).count()
+
+    @classmethod
+    def from_bank_account_number(cls, session, bank_account_number):
```
Comment (djpnewton) [inline models.py]: change to "from_account_number"

File: models.py [hunk 2]
```diff
@@ -378,6 +378,38 @@ def to_json(self):
...
+        return session.query(cls).filter(cls.account_number == account_number).first()
+
+    def __repr__(self):
+        return "<Bank Account Number %r>" % (self.account_number)
```
Comment (djpnewton) [inline models.py]: Change to "<Bank %r>"

**Comment by djpnewton** [comment]
add "bank_name"

File: models.py (hunk 2 above)
Comment (eoliveros) [reply models.py]: Should be changed.

**Comment by eoliveros** [comment]
> add "bank_name"

Added bank_name.

File: admin.py [hunk 3]
```diff
@@ -22,6 +22,7 @@
 admin.add_view(MerchantTxRestrictedModelView(MerchantTx, db.session, category='Reports'))
 admin.add_view(SettlementAdminModelView(Settlement, db.session, category='Reports'))
 admin.add_view(SettlementRestrictedModelView(Settlement, db.session, endpoint='SettlementUser', category='Reports'))
+admin.add_view(BankRestrictedModelView(Bank, db.session))
```
Comment (djpnewton) [inline admin.py]: put this in the 'Users' category, group it with the others in that category

**Comment by djpnewton** [comment]
We need a 'default' (bool) field as well

File: admin.py (hunk 3 above)
Comment (eoliveros) [reply admin.py]: Added to users categorry and rearranged admin.py

**Comment by eoliveros** [comment]
> We need a 'default' (bool) field as well
Added account_status with boolean.

**Comment by eoliveros** [comment]
> > We need a 'default' (bool) field as well
> > Added account_status with boolean.

actually renaming it to default_account

**Comment by djpnewton** [comment]
is it ready for me to test?

**Comment by eoliveros** [comment]
> is it ready for me to test?

should be, its in the latest push i did.

**Comment by eoliveros** [comment]
the column for the default still pointing to the old one, so i pushed another one to fix it.

**Pull request merged**